### ISSUE TITLE
Inventory assigns "yes" and "no" as strings and not as booleans

### ISSUE THREAD

**Issue by fthommen** [ansible/ansible#25778]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

Inventory

##### ANSIBLE VERSION

```
ansible 2.3.0.0
  [...]
  configured module search path = Default w/o overrides
  python version = 2.7.5 (default, Nov  6 2016, 00:28:07) [GCC 4.8.5 20150623 (Red Hat 4.8.5-11)]

```

##### OS / ENVIRONMENT
CentOS 7.x on x86_64

##### SUMMARY
Even though ansible uses "yes" and "no" as boolean values, a variable set to "yes" or "no" in the inventory seems to be seen as string and not as boolean.

##### STEPS TO REPRODUCE

- Create ansible with entries

```
host1 doecho=true
host2 doecho=yes
```
- Run this playbook on these hosts:

```yaml
---
- hosts: all
  tasks:
  - shell: "echo \"I am here\""
    register: echo
    when: doecho

  - debug:
      var: echo
```

##### EXPECTED RESULTS
Playbook succeeds on both hosts

##### ACTUAL RESULTS
The playbook succeeds on host1 and fails on host2 with error "The conditional check 'doecho' failed. The error was: error while evaluating conditional (doecho): 'yes' is undefined"

This is inconsistent with the handling of "yes" and "no" within playbooks.

**Comment by pilou-** [comment]
Variables which are not [behavioral inventory parameters](https://docs.ansible.com/ansible/intro_inventory.html#list-of-behavioral-inventory-parameters) and are defined in an INI-like inventory are interpreted as strings. This doesn't seem [documented](https://docs.ansible.com/ansible/intro_inventory.html), there is an hint in the note [here](https://docs.ansible.com/ansible/playbooks_variables.html#passing-variables-on-the-command-line). A not empty string will be evaluated to 
```
$ ansible localhost -e var=false -m debug -a 'msg="{{ var | default('d', true) }}"'
[...]
"msg": "false"
[...]
$ ansible localhost -e var=no -m debug -a 'msg="{{ var | default('d', true) }}"'
[...]
"msg": "no"
[...]
```

There is a bug related to evaluation of `when` clause when this clause contains only a variable name.

**Comment by fthommen** [comment]
very pity and IMHO somehow inconsistent.  In this case - since that seems to be a design decision - please make this issue a documentation issue, not a bug report.

**Comment by bcoca** [comment]
fixed by #25798
